cores/esp8266/core_esp8266_main.cpp

```diff
@@ -408,71 +409,97 @@ uint32_t __flashindex;
+    extern void set_pll(void);
+    set_pll();  // fix printing for 115200 bps
+    ets_uart_printf(PSTR("\n\nConfig info referenced for system partition table registration:\n"));
+    ets_uart_printf(PSTR("  %-18s 0x%08X\n"), PSTR("PHY_DATA"), phy_data);
+    ets_uart_printf(PSTR("  %-18s 0x%08X\n"), PSTR("RF_CAL"), rf_cal);
+    ets_uart_printf(PSTR("  %-18s 0x%08X\n"), PSTR("SYSTEM_PARAMETER"), system_parameter);
+    ets_uart_printf(PSTR("  %-18s 0x%08X %u\n"), PSTR("chip_size"), flashchip->chip_size, flashchip->chip_size);
+    ets_uart_printf(PSTR("  %-18s 0x%08X\n"), PSTR("EEPROM_start"), EEPROM_start);
+    ets_uart_printf(PSTR("  %-18s 0x%08X\n"), PSTR("FS_start"), FS_start);
+    ets_uart_printf(PSTR("  %-18s 0x%08X\n"), PSTR("FS_end"), FS_end);
+    ets_uart_printf(PSTR("  %-18s 0x%08X\n"), PSTR("FS_page"), FS_page);
+    ets_uart_printf(PSTR("  %-18s 0x%08X\n"), PSTR("FS_block"), FS_block);
+#if !defined(FLASH_MAP_SUPPORT)
+    uint32_t configured_chip_size = system_parameter + 4096 * 3;
+    // flashchip->chip_size is updated by the SDK. The size is based on the
+    // value patched into the .bin header by esptool.
+    // system_get_flash_size_map() returns that patched value.
+    if (flashchip->chip_size != configured_chip_size)
+    {
+      // For this message and postmortem to be readable, the console speed may
+      // need to be 74880 bps.
+      ets_uart_printf(PSTR("\nMissmatch between actual(%u) vs configured(%u) flash size\n"), flashchip->chip_size, configured_chip_size);
+      // Full stop to avoid possible stored flash data corruption. This
+      // missmatch does not occur with flash size selection "Mapping defined by
+      // Hardware and Sketch".
+      abort();
+    }
 #endif
+#endif
+    if (!system_partition_table_regist(at_partition_table, sizeof(at_partition_table) / sizeof(at_partition_table[0]), system_get_flash_size_map()))
+    {
+      // We will land here anytime the build flash size value set by the Arduino
+      // IDE Tools menu mismatches the firmware image value detected and updated
+      // on the fly by esptool.
+      //
+      // The SDKs PLL CPU clock calibration hasn't run. For this message and
+      // postmortem to be readable, the console speed may need to be 74880 bps.
+      //
+      // Because SDK v3.0.x always has a non-32-bit wide exception handler
+      // installed, we can use PROGMEM strings with Boot ROM print functions.
+      ets_uart_printf(PSTR("\nSystem partition table registration failed!\n"));
+
+      //C What is the best action on failure?
+      //C * The above printf maybe redundant. `system_partition_table_regist`
+      //C   appears to print specific failure messages. most of the time.
+      //C * Things will not get better by rebooting.
+      //C * Most likely the error message was not readable - mismatched console speed
+      //C * Expressif example do a `while(true){};` on fail. HWDT appears to be
+      //C   off and the device freezes.
+      abort();
```

Comment:
What is the preferred action for `system_partition_table_regist` failure?
* The above printf may be redundant. `system_partition_table_regist`
  appears to print specific failure messages. most of the time.
* Things will not get better by rebooting.
* Most likely the error message was not readable - mismatched console speed
* Expressif example do a `while(true){};` on fail. HWDT appears to be
  off and the device freezes.

Reply:
Preferably, we should be able to handle partition(s) size and address and also validate them at build time to avoid this path altogether. But, for this set of choices, while(true) loop seems most sensible here.

Reply:
My final solution was to force the printing speed to 11500bps. (special case) Print messages, block on input character read in a `do {} while(true);` loop. Message repeats on any character received.